spec/Api/ReferenceEntityApiSpec.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+namespace spec\Akeneo\PimEnterprise\ApiClient\Api;
+
+use Akeneo\Pim\ApiClient\Client\ResourceClientInterface;
+use Akeneo\Pim\ApiClient\Pagination\PageFactoryInterface;
+use Akeneo\Pim\ApiClient\Pagination\PageInterface;
+use Akeneo\Pim\ApiClient\Pagination\ResourceCursorFactoryInterface;
+use Akeneo\Pim\ApiClient\Pagination\ResourceCursorInterface;
+use Akeneo\PimEnterprise\ApiClient\Api\ReferenceEntityApi;
+use Akeneo\PimEnterprise\ApiClient\Api\ReferenceEntityApiInterface;
+use PhpSpec\ObjectBehavior;
+
+class ReferenceEntityApiSpec extends ObjectBehavior
+{
+    function let(
+        ResourceClientInterface $resourceClient,
+        PageFactoryInterface $pageFactory,
+        ResourceCursorFactoryInterface $cursorFactory
+    ) {
+        $this->beConstructedWith($resourceClient, $pageFactory, $cursorFactory);
+    }
+
+    function it_is_a_reference_entity_api()
+    {
+        $this->shouldImplement(ReferenceEntityApiInterface::class);
+    }
+
+    function it_returns_a_reference_entity(ResourceClientInterface $resourceClient)
+    {
+        $referenceEntity = [
+            'code' => 'designer',
+            'values' => [
```

Comment:
It's an example of a record. A reference entity doesn't have values. Same for `upsert` and `upsertList`